### PR TITLE
Config parameters for kick evaluator modified

### PR DESCRIPTION
**Pull request by Yudai-8-om**
## Description
Modified configuration parameters for ros2 kick evaluator
Updated window evaluator

## Associated Issue
Issue #1600

### CODE REVIEW

File: soccer/src/soccer/kick_evaluator.hpp [hunk 1]
```diff
@@ -10,6 +11,13 @@
 #include "optimization/parallel_gradient_1d_config.hpp"
 #include "optimization/parallel_gradient_ascent_1d.hpp"
 
+constexpr auto kickEvaluatorParamModule = "kick_evaluator";
```
Comment (kylestach) [inline soccer/src/soccer/kick_evaluator.hpp]: Nit: `kKickEvaluatorParamModule`

File: soccer/src/soccer/kick_evaluator.hpp [hunk 2]
```diff
@@ -100,7 +108,7 @@ class KickEvaluator {
      */
     std::vector<rj_geometry::Point> hypothetical_robot_locations;
 
-    static ConfigDouble* kick_std_dev;
+    // DECLARE_NS_FLOAT64(kickEvaluatorParamModule, kick_evaluator, kick_std_dev)
```
Comment (kylestach) [inline soccer/src/soccer/kick_evaluator.hpp]: You can just delete this line.

File: soccer/src/soccer/kick_evaluator.hpp [hunk 3]
```diff
@@ -141,7 +149,8 @@ class KickEvaluator {
         const std::vector<float>& robot_st_devs, float boundary_lower,
         float boundary_upper);
 
-    static ConfigDouble* kick_mean;
-    static ConfigDouble* robot_std_dev;
-    static ConfigDouble* start_x_offset;
+    
+ //    DECLARE_NS_FLOAT64(kickEvaluatorParamModule, kick_evaluator, kick_mean)
+	// DECLARE_NS_FLOAT64(kickEvaluatorParamModule, kick_evaluator, robot_std_dev)
+	// DECLARE_NS_FLOAT64(kickEvaluatorParamModule, kick_evaluator, start_x_offset)
```
Comment (kylestach) [inline soccer/src/soccer/kick_evaluator.hpp]: Delete these too

File: soccer/src/soccer/kick_evaluator.cpp [hunk 4]
```diff
@@ -102,8 +106,8 @@ KickResults KickEvaluator::eval_pt_to_seg(Point origin, Segment target) {
     // Create function with only 1 input
     // Rest are bound to constant values
     function<tuple<float, float>(float)> ke_func =
-        bind(&eval_calculation, std::placeholders::_1, (kick_mean->value()),
-             (kick_std_dev->value()), cref(bot_means), cref(bot_st_devs), cref(bot_vert_scales),
+        bind(&eval_calculation, std::placeholders::_1, kick_evaluator::PARAM_kick_mean,
+             kick_evaluator::PARAM_kick_std_dev, cref(bot_means), cref(bot_st_devs), cref(bot_vert_scales),
              target_width / -2, target_width / 2);
```
Comment (kylestach) [inline soccer/src/soccer/kick_evaluator.cpp]:
```suggestion
        [] (float x) {
            return eval_calculation(x,
                                    kick_evaluator::PARAM_kick_mean,
                                    kick_evaluator::PARAM_kick_std_dev,
                                    cref(bot_means),
                                    cref(bot_st_devs),
                                    cref(bot_vert_scales),
                                    -target_width / 2,
                                    target_width / 2);
```

**Review by kylestach — CHANGES_REQUESTED** [review]
Pretty good, a few minor suggestions (plus a fix for the failing clang-tidy check)

File: soccer/src/soccer/kick_evaluator.cpp (hunk 4 above)
Comment (Yudai-8-om) [reply soccer/src/soccer/kick_evaluator.cpp]: This code raised an error, so I added & inside the bracket(not sure that's the right way to fix it).

File: soccer/src/soccer/kick_evaluator.cpp [hunk 5]
```diff
@@ -28,10 +28,7 @@ inline double fast_exp(double x) {
 inline float fast_exp(float x) { return static_cast<float>(fast_exp(static_cast<double>(x))); }
 
 void KickEvaluator::create_configuration(Configuration* cfg) {
```
Comment (kylestach) [inline soccer/src/soccer/kick_evaluator.cpp]: You can delete this function completely.

Comment (kylestach) [reply soccer/src/soccer/kick_evaluator.cpp]: Also, delete the `REGISTER_CONFIGURABLE` on line 10.

**Review by kylestach — APPROVED** [review]
A few minor changes and it should be good!

**Review by HussainGynai — APPROVED** [review]
Looks good to me, just fix what Kyle has said

**Pull request merged**